src/wp-includes/script-loader.php

```diff
@@ -2869,7 +2869,7 @@ function wp_enqueue_editor_format_library_assets() {
  * @return string String made of sanitized `<script>` tag attributes.
  */
 function wp_sanitize_script_attributes( $attributes ) {
-	$html5_script_support = ! is_admin() && ! current_theme_supports( 'html5', 'script' );
+	$html5_script_support = is_admin() || current_theme_supports( 'html5', 'script' );
```

Comment:
So essentially this variable should previously been called `$not_html5_script_support`. In other words, the naming was inverted.